src/http.c

```diff
@@ -957,6 +1015,43 @@ http_file_rewind(struct http_file *file)
 }
 
 void
+http_populate_cookies(struct http_request *req)
+{
+	int						 i, v;
+	size_t					 nlen;
+	struct http_cookie		*ck;
+	char					*c, *header, *value;
+	char					*cookies[HTTP_MAX_COOKIES];
+
+	if (!http_request_header(req, "cookie", &c))
+	   return;
+
+	header = kore_strdup(c);
+	v = kore_split_string(header, ";", cookies, HTTP_MAX_COOKIES);
+	for (i = 0; i < v; i++) {
+		for (c = cookies[i]; isspace(*c); c++)
+			;
+
+		if ( (value = strchr(c, '=')) == NULL) {
+			kore_log(LOG_ERR, "malformed cookie");
+			continue;
+		}
+		nlen = value - c - 1;
+		if (nlen > HTTP_MAX_COOKIENAME) {
+			kore_log(LOG_ERR, "cookie name is too long");
+			continue;
+		}
+		value = "\0";
```

Comment:
This seems very wrong.

You are pointing value to an empty C string which is then used in kore_strdup() for the name member.

After that you increase the pointer by one position (value++) and store whatever value is pointing towards now in the value member using kore_strdup() again.

I guess you ment something more like:

```
*(value)++ = '\0';
ck->name = kore_strdup(c);
ck->value = kore_strdup(value);
```